src/lib/optimizer/strategy/stored_table_column_alignment_rule.cpp

```diff
@@ -7,13 +7,19 @@
 #include <vector>
 
 #include "logical_query_plan/abstract_lqp_node.hpp"
+#include "logical_query_plan/logical_plan_root_node.hpp"
 #include "logical_query_plan/lqp_utils.hpp"
+#include "logical_query_plan/stored_table_node.hpp"
 
 namespace {
+using namespace opossum;  // NOLINT
+
 // Modified hash code generation for StoredTableNodes where column pruning information is omitted. Struct is used to
 // enable hash-based containers containing std::shared_ptr<StoredTableNode>.
 struct StoredTableNodeSharedPtrHash final {
   size_t operator()(const std::shared_ptr<opossum::StoredTableNode>& node) const {
+    DebugAssert(std::is_sorted(node->pruned_chunk_ids().cbegin(), node->pruned_chunk_ids().cend()),
+                "Expected sorted vector of ChunkIDs");
```

Comment:
As much as I appreciate you adding assertions, this is already guaranteed in `set_pruned_column_ids`. (multiple places)